hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ServerCrashProcedure.java

```diff
@@ -466,20 +466,31 @@ private void assignRegions(MasterProcedureEnv env, List<RegionInfo> regions) thr
       RegionStateNode regionNode = am.getRegionStates().getOrCreateRegionStateNode(region);
       regionNode.lock();
       try {
+        // This is possible, as when a server is dead, TRSP will fail to schedule a RemoteProcedure
+        // to us and then try to assign the region to a new RS. And before it has updated the region
+        // location to the new RS, we may have already called the am.getRegionsOnServer so we will
+        // consider the region is still on us. And then before we arrive here, the TRSP could have
+        // updated the region location, or even finished itself, so the region is no longer on us
+        // any more, we should not try to assign it again. Please see HBASE-23594 for more details.
+        if (!serverName.equals(regionNode.getRegionLocation())) {
+          LOG.info("{} found a region is no longer on us {} but on {}, give up assigning...", this,
```

Comment:
Fix this log message. Do you mention the region name? You should?  Just print out the full regionNode.